devicetypes/qubino/qubino-3-phase-meter.src/qubino-3-phase-meter.groovy

```diff
@@ -0,0 +1,194 @@
+/**
+ *  Copyright 2020 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "Qubino 3 Phase Meter", namespace: "qubino", author: "SmartThings", ocfDeviceType: "x.com.st.d.energymeter", mcdSync: true) {
+		capability "Energy Meter"
+		capability "Power Meter"
+		capability "Configuration"
+		capability "Sensor"
+		capability "Health Check"
+		capability "Refresh"
+
+		fingerprint mfr: "0159", prod: "0007", model: "0054", deviceJoinName: "Qubino Energy Monitor" //Qubino 3 Phase Meter
+	}
+
+	// tile definitions
+	tiles(scale: 2) {
+		multiAttributeTile(name:"power", type: "generic", width: 6, height: 4){
+			tileAttribute("device.power", key: "PRIMARY_CONTROL") {
+				attributeState("default", label:'${currentValue} W')
+			}
+			tileAttribute("device.energy", key: "SECONDARY_CONTROL") {
+				attributeState("default", label:'${currentValue} kWh')
+			}
+		}
+		standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat",width: 2, height: 2) {
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+		standardTile("configure", "device.power", inactiveLabel: false, decoration: "flat",width: 2, height: 2) {
+			state "configure", label:'', action:"configuration.configure", icon:"st.secondary.configure"
+		}
+
+		main (["power","energy"])
+		details(["power","energy","refresh", "configure"])
+	}
+}
+
+def installed() {
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	state.numberOfMeters = 3
```

Comment:
Since it's only used here, does numberOfMeters have to be kept in state ?

Reply:
My goal was to introduce some part of flexibility, in case similar device with different number of endpoints has to be integrated